src/bp-activity/classes/class-bp-activity-activity.php

```diff
@@ -1667,7 +1667,18 @@ public static function get_activity_comments( $activity_id, $left, $right, $spam
 
 				$sql = "{$sql['select']} {$sql['from']} {$sql['where']} {$sql['misc']}";
 
-				$sql = $wpdb->prepare( $sql, $top_level_parent_id, $left, $right );
+				/**
+				 * Filters the MySQL prepared statement for the activity query.
+				 *
+				 * @since BuddyBoss 2.4.1
+				 *
+				 * @param string $value       Prepared statement for the activity query.
+				 * @param int    $activity_id Activity ID to fetch comments for.
+				 * @param int    $left        Left-most node boundary.
+				 * @param int    $right       Right-most node boundary.
+				 * @param string $spam_sql    SQL Statement portion to differentiate between ham or spam.
+				 */
+				$sql = apply_filters( 'bp_activity_comments_user_join_filter', $wpdb->prepare( $sql, $top_level_parent_id, $left, $right ), $activity_id, $left, $right, $spam_sql );
```

Comment:
@raftaar1191 this filter is used for the legacy version because they have a user table join.
check here: https://github.com/buddyboss/buddyboss-platform/pull/4063/files#diff-4a6c0ee301c9b9b7b3ec6042c727f18d88962b1fd7fd09e69e984d8102832444R1597

Newer the version does not have a user table join.

Also, we have a join SQL here if you needed.
https://github.com/buddyboss/buddyboss-platform/pull/4063/files#diff-4a6c0ee301c9b9b7b3ec6042c727f18d88962b1fd7fd09e69e984d8102832444R1656

Reply:
Thanks @KartikSuthar